ament_lint_auto/cmake/ament_lint_auto_find_test_dependencies.cmake

```diff
@@ -21,6 +21,7 @@
 # @public
 #
 macro(ament_lint_auto_find_test_dependencies)
+  cmake_parse_arguments(ARG "" "" "EXCLUDE" ${ARGN})
```

Comment:
Since this is a macro please prefix all "local" variables with an underscore: `_`.

The new option also needs to be documented in the docblock.

Doesn't the following check have to be remove to make it even continue when `EXCLUDE` is passed? Have you tried to use this patch with the new `EXCLUDE` option? I would suggest to also update at least one of the existing cases where we skip a specific linter by setting the `FOUND` flag to `FALSE` before to use this new option to demonstrate this works.

Reply:
@dirk-thomas Did you mean "...where we skip a specific linter by setting the FOUND flag to **TRUE**"? e.g.  https://github.com/ament/ament_index/blob/master/ament_index_cpp/CMakeLists.txt#L43

---

The check doesn't fail, but you are right that I should check for ARG_UNPARSED_ARGUMENTS. It turns out that macro arguments are not variables which means they don't work with `if()` (https://cmake.org/cmake/help/v3.0/command/if.html):

> `if(<variable>)`
    True if the variable is defined to a value that is not a false constant. False otherwise. (Note macro arguments are not variables.)

The following would've caught the error we are expecting:

```cmake
set(FOO ${ARGN})
if(FOO)
  message(FATAL_ERROR ...)
endif()
```